tests/enum/tenumunsignedconverter.nim

```diff
@@ -0,0 +1,23 @@
+#issue #20024
```

Comment:
This test does not work within the "megatest" environment. Please investigate.

Reply:
Do you know how I can reproduce the environment where this fails? The test works for me locally. It looks like I need to compile for i386 to debug, but I'm on Windows 10.

Reply:
Do you know how to get more info on the failure? 
https://dev.azure.com/nim-lang/Nim/_build/results?buildId=21882&view=logs&j=ba7bbfa4-f55c-5c34-6d52-1b6b4edd3f37&t=01b6eca1-ed82-5532-a6ed-8d9a0cd0188a&l=3372

Is there a way to see what's produced in the nimcache?

 I was able to compile a 32-bit executable via WSL:
```
genix@PC-MUGEN:~/nimdev$ file enumbug
enumbug: ELF 32-bit LSB shared object, Intel 80386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.2, BuildID[sha1]=f82ef941f3dcd141a6af19191cf1c6b57dba325d, for GNU/Linux 3.2.0, not stripped
```
But the test passes.

Reply:
In my nim.cfg:
```
nimcache = ".nimcache"
@if i386:
  cpu = i386
  passC = "-m32"
  passL = "-m32"
@end
```
compiled with:
```
genix@PC-MUGEN:~/nimdev$ nim c -f -d:i386 enumbug.nim
Hint: used config file '/home/genix/Nim/config/nim.cfg' [Conf]
Hint: used config file '/home/genix/Nim/config/config.nims' [Conf]
Hint: used config file '/home/genix/nimdev/nim.cfg' [Conf]
.................................................................
/home/genix/nimdev/enumbug.nim(18, 38) Warning: conversion to enum with holes is unsafe: Foo(a) [HoleEnumConv]
CC: ../Nim/lib/std/private/digitsutils.nim
CC: ../Nim/lib/std/assertions.nim
CC: ../Nim/lib/system/dollars.nim
CC: ../Nim/lib/pure/collections/sharedlist.nim
CC: ../Nim/lib/system.nim
CC: enumbug.nim
Hint:  [Link]
Hint: mm: refc; threads: on; opt: none (DEBUG BUILD, `-d:release` generates faster code)
28747 lines; 0.396s; 31.461MiB peakmem; proj: /home/genix/nimdev/enumbug.nim; out: /home/genix/nimdev/enumbug [SuccessX]
genix@PC-MUGEN:~/nimdev$ ./enumbug 
```
Without error.

Reply:
I'm currently trying to install 32-bit Ubuntu-18.04 via virtualbox.

Reply:
I got Ubuntu 32-bit installed via virtualbox.  When I tried to compile the test I got the error at compile time, so this is a different issue.

I tried compiling this:
```nim
type Foo = enum
  A = 0x8000_0000
```
which fails with: `Error: unhandled exception: value out of range: 2147483648 not in -2147483648 .. 2147483647 [RangeDefect]`

Reply:
For now, I've updated the test to only run when the platform is 64-bit. I did this by `when pointer.sizeof == 8`. Let me know if there's a better option. I'll create a new issue for the other bug.

Reply:
The issue might be the converters. Wrap the entire test in `block:`. Otherwise the converters leak into the other tests.